util/src/main/scala/org/ensime/util/Debouncer.scala

```diff
@@ -0,0 +1,118 @@
+// Copyright: 2010 - 2017 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.util
+
+import akka.actor.Scheduler
+import java.time.Clock
+import java.util.concurrent.atomic.{ AtomicBoolean, AtomicInteger }
+import org.slf4j.LoggerFactory
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+/**
+ * Generic debouncer implementation
```

Comment:
it is kind of weird that this isn't an `Actor` since we use akka everywhere else (I'd prefer to use fs2). Can you please add a comment explaining that writing this as an akka actor has performance problems OR (if you don't think that is completely correct) create a ticket to rewrite this as an actor. Perhaps if we were to disable logging for the debouncer it might be ok.